File: frontend/types/booking.ts
export type Booking = {
  bookingId: string;
  tenantId: string;
  businessSlug: string;
  productId?: number;
  serviceName: string;
  customerName: string;
  customerEmail: string;
  customerPhone: string;
  bookingDate: string;
  startTime: string;
  endTime: string;
  status: string;
  notes?: string;
};

export type AvailabilitySlot = {
  startTime: string;
  endTime: string;
  available: boolean;
};

File: booking/src/main/java/com/vitral/booking/domain/model/Booking.java
package com.vitral.booking.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Booking {
    private String bookingId;
    private String tenantId;
    private String businessSlug;
    private Long productId;
    private String serviceName;
    private String customerName;
    private String customerEmail;
    private String customerPhone;
    private LocalDate bookingDate;
    private LocalTime startTime;
    private LocalTime endTime;
    private String status;
    private String notes;
    private LocalDateTime createdAt;
}
